libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -463,89 +463,35 @@ func (dhtClient *DHTClient) RouteEnvelope(envel *aea.Envelope) error {
 		Str("target", target).
 		Msg("requesting agent record from relay...")
 
-	lookupRequestPerformative := &acn.LookupRequestPerformative{AgentAddress: target}
-	msg := &acn.AcnMessage{
-		Performative: &acn.LookupRequest{LookupRequest: lookupRequestPerformative},
-	}
-	buf, err := proto.Marshal(msg)
-	if err != nil {
-		lerror(err).
-			Str("op", "route").
-			Str("target", target).
-			Msg("while serializing LookupRequest message")
-		errReset := stream.Reset()
-		ignore(errReset)
-		return err
-	}
-	err = utils.WriteBytes(stream, buf)
-	if err != nil {
-		lerror(err).
-			Str("op", "route").
-			Str("target", target).
-			Msg("while sending LookupRequest message")
-		errReset := stream.Reset()
-		ignore(errReset)
-		return err
-	}
+	streamPipe := utils.StreamPipe{Stream: stream}
 
-	buf, err = utils.ReadBytes(stream)
+	err = acn.SendLookupRequest(streamPipe, target)
 	if err != nil {
 		lerror(err).
 			Str("op", "route").
 			Str("target", target).
-			Msg("while receiving response acn message")
+			Msg("while sending LookupRequest message")
 		errReset := stream.Reset()
 		ignore(errReset)
 		return err
 	}
 
-	response := &acn.AcnMessage{}
-	err = proto.Unmarshal(buf, response)
+	record, err := acn.ReadLookupResponse(streamPipe)
```

Comment:
Given they are always used together, why not wrap it into a method in utils?

Reply:
nice point